local_php.ini

```diff
@@ -7,6 +7,15 @@
 ; Disable aggressive caching
 opcache.enable = 0
 
+; Enable xdebug
+; https://enrise.com/2018/02/debugging-php-with-xdebug/
+zend_extension=/app/.heroku/php/lib/php/extensions/no-debug-non-zts-20131226/xdebug.so
+xdebug.remote_enable = On
+xdebug.remote_connect_back = Off
+xdebug.remote_host = host.docker.internal
```

Comment:
@DatGuyJonathan - This is special for Mac.  See https://docs.docker.com/docker-for-mac/networking/

We need to do something special for Linux's Docker. Open to suggestions.

Reply:
https://github.com/docker/for-linux/issues/264#issuecomment-410048049 is probably useful here.

Reply:
That proposal works for me. Other solutions I found involved passing around environment variables, so they aren't as straightforward. Here's what I have for my Linux:

```
diff --git a/docker-compose.yml b/docker-compose.yml
index 661734e..fa44344 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -2,7 +2,7 @@ version: '3'
 services:
   web:
     build: .
-    command: vendor/bin/heroku-php-apache2 -F fpm_custom_local.conf -i local_php.ini workbench
+    entrypoint: /app/docker-entrypoint.sh
     volumes:
       - './workbench:/app/workbench'
     environment:
diff --git a/docker-entrypoint.sh b/docker-entrypoint.sh
new file mode 100755
index 0000000..1ad686a
--- /dev/null
+++ b/docker-entrypoint.sh
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+function fix_linux_internal_host() {
+  DOCKER_INTERNAL_HOST="host.docker.internal"
+
+  if ! grep $DOCKER_INTERNAL_HOST /etc/hosts > /dev/null ; then
+    DOCKER_INTERNAL_IP=`/sbin/ip route | awk '/default/ { print $3 }' | awk '!seen[$0]++'`
+    echo -e "$DOCKER_INTERNAL_IP\t$DOCKER_INTERNAL_HOST" | tee -a /etc/hosts > /dev/null
+    echo "Added $DOCKER_INTERNAL_HOST to hosts /etc/hosts"
+  fi
+}
+
+fix_linux_internal_host
+vendor/bin/heroku-php-apache2 -F fpm_custom_local.conf -i local_php.ini workbench
\ No newline at end of file
```

Reply:
Thanks, looks reasonable. Could you add that to this branch? Also, be sure to add docker-entrypoint.sh to the .slugignore.

Reply:
:+1:

Reply:
Hold please, I think this broke Mac. I'm looking into it.

Reply:
Mac couldn't debug with that `fix_linux_internal_host` because the container had `host.docker.internal` in its /etc/hosts. Only Linux needs that, so I needed to change the script to only execute if the host OS is Linux.

I brought in `iputils-ping` to ping `host.docker.internal`. On Mac, that will succeed and return error code 0, so the Linux workaround will not execute. On Linux, the ping fails and returns error code 2, so the workaround executes.

The container build doesn't seem to take that much longer, but let me know if we want to avoid ping. The alternative is to use `uname` in local.sh, pass that to docker-compose, set it as an environment variable in the container, then the entrypoint script will validate the result of `uname` is Linux.

Reply:
Are you able to use `host` for this? I was doing some debugging yesterday and remember that `host` is already part of the image.

So, `host host.docker.internal` and use the output of that.